### PR TITLE
feat(novui,web): General UI / UX fixes and improvements

### PR DESCRIPTION
**Pull request by antonjoel82**
### What changed? Why was the change needed?

- Little styling tweaks
- Change buttons to match [DS spec](https://www.figma.com/design/vmeumrFqggt6cUpPQwJgr4/DS?m=dev&node-id=206-1497)
- Conditionally show Sync button (not functional yet)

### Screenshots

Screenshot 2024-06-20 at 10 33 13 PM
Screenshot 2024-06-20 at 10 34 22 PM

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]

**Review by scopsy — APPROVED** [review]

File: apps/web/src/components/layout/components/v2/HeaderNav.tsx [hunk 1]
```diff
@@ -33,6 +38,12 @@ export function HeaderNav() {
     >
       {/* TODO: Change position: right to space-between for breadcrumbs */}
       <HStack flexWrap={'nowrap'} justifyContent="flex-end" gap={'100'}>
+        {/* TODO: Use a more robust / realistic way to conditionally show this button */}
```
Comment (denis-kralj-novu) [inline apps/web/src/components/layout/components/v2/HeaderNav.tsx]: ❓ **question:** ‏ From my understanding, this button should show up on all environments (Local, Dev, Prod) once v2 is released. I would imagine hiding it behind a feature flag would be the correct choice no?

Comment (scopsy) [reply apps/web/src/components/layout/components/v2/HeaderNav.tsx]: Yes @denis-kralj-novu great catch! Joel is out today, do you mind adding to this PR so we can merge it?

Comment (denis-kralj-novu) [reply apps/web/src/components/layout/components/v2/HeaderNav.tsx]: Sure. I've already used the flag in my PR that is handling the logic for the top buttons so I know what to do

Comment (denis-kralj-novu) [reply apps/web/src/components/layout/components/v2/HeaderNav.tsx]: Done 🚀

**Pull request merged**